fix: add task input param annotations to support programmatic documentation

saugupta1 opened the pull request:
For VCP CLI improvements (VC-4028 and VC-4517, which are not publicly visible). Allows the VCP CLI to programmatically generate CLI help documentation strings for task params. Also allows the CLI to introspect the baseline input params.

- Updated Task Registry for better introspection on available task making it easier to create cli
- Enhanced Task Input Models
- Added Task Baseline inputs to separate out baseline args.

atolopko-czi commented on src/czbenchmarks/cli/cli.py:
```diff
@@ -4,7 +4,7 @@
 from .cli_list import list_cmd
 
-# from .cli_run import run # Run command not exposed yet
+# from .cli_run import run
```
why remove the comment? it's still disabled.

atolopko-czi commented on src/czbenchmarks/tasks/single_cell/perturbation_expression_prediction.py:
```diff
@@ -1,33 +1,73 @@
-    Dataclass to contain input parameters for the PerturbationExpressionPredictionTask.
-    The row and column ordering of the model predictions can optionallybe provided as
-    cell_index and gene_index, respectively, so the task can align a model matrix that
-    is a subset of or re-ordered relative to the dataset adata.
```
Reinstate

atolopko-czi reviewed (changes requested):
Why make all the changes to the `cli` module if we're not providing the `run` command in the CLI? I don't believe any of this code is being used by the VCP CLI, so are these changes strictly needed? I'm concerned we're trying to maintain similar functionality across two repositories, which is going to be hard to do. I feel we should remove this code, but I want to first understand if there are potential justifications for keeping it.

Aside from that, there are also a few comments  to consider or address.

saugupta1 commented:
Some of the changes are due to cursor having its own mind. Will fix them when I receive all of the review comments.

saugupta1 commented:
> Why make all the changes to the `cli` module if we're not providing the `run` command in the CLI? I don't believe any of this code is being used by the VCP CLI, so are these changes strictly needed? I'm concerned we're trying to maintain similar functionality across two repositories, which is going to be hard to do. I feel we should remove this code, but I want to first understand if there are potential justifications for keeping it.
> 
> Aside from that, there are also a few comments to consider or address.

For easier testing of tasks using cli rather than switching between repo. We can get rid of it when desired as it is anyway unused code.

saugupta1 commented:
### General Overview

- **Type annotations and Pydantic integration:**  
  All task input classes now use `Annotated` types with `pydantic.Field` for richer input validation and metadata descriptions, improving clarity for API users and downstream tooling.
- **Input validation:**  
  Many task input classes now include `@field_validator` methods to enforce type and value constraints (e.g., positive integers, list-like objects).
- **Baseline model support:**  
  Each task class now defines a `baseline_model` attribute referencing a baseline input class, standardizing how baseline comparisons are handled.
- **Logging improvements:**  
  Extensive debug-level logging has been added across tasks to track key shapes, parameters, and computation steps, aiding in diagnostics and transparency.
- **Sparse matrix handling:**  
  Tasks that process embeddings or gene expression data now consistently convert sparse matrices to dense arrays before metric computation for compatibility.
- **Minor fixes and docstring updates:**  
  Several minor code, docstring, and error message improvements for clarity and correctness.

---

### File-by-File Summary

#### clustering.py
- Task input uses `Annotated` fields with descriptions and validators.
- Added validation for `n_iterations` to be positive.
- More debug logging for shapes and parameters.
- Added `baseline_model` attribute.

#### embedding.py
- Task input uses `Annotated` fields for labels.
- Added `baseline_model` and debug logging.
- Converts sparse matrices to dense before metrics.

#### integration.py
- Task input uses `Annotated` fields with descriptions and validators for `batch_labels` and `labels`.
- Added `baseline_model`, improved logging, and sparse-to-dense conversion.

#### label_prediction.py
- Task input uses `Annotated` fields and validators for `labels`, `n_folds`, and `min_class_size`.
- Added `LabelPredictionBaselineInput` and set `baseline_model`.
- Improved pipeline configuration for classifiers (scalers now use `with_mean=False`).
- Baseline computation now explicitly handles sparse matrices.

#### sequential.py
- Task input uses `Annotated` fields and validators for all parameters.
- Added `baseline_model`, improved logging, and sparse-to-dense conversion.

#### single_cell/cross_species_integration.py
- Task input uses `Annotated` fields and validators.
- Added `baseline_model`.
- Updated logging and error messages.
- Baseline computation method signature standardized.

#### single_cell/cross_species_label_prediction.py
- Task input uses `Annotated` fields and validators.
- Added `baseline_model` and improved logging.
- Minor fixes to docstrings and error messages.
- Baseline computation method signature standardized.

#### single_cell/perturbation_expression_prediction.py
- Task input uses `Annotated` fields and validators for all parameters.
- Added `baseline_model`.
- Improved logging and error messages.
- Baseline computation method signature standardized.

mlgill commented:
> > I would like to see what the impact is to the integration tests and the example scripts. @saugupta1 @atolopko-czi Can you ping me when those are updated? Thanks
> 
> I'm not sure we need to update any of these, in fact. @saugupta1 What are your thoughts?

Assuming the impact to these higher level workflows is minimal-to-none, then I think this is an improvement. Approved from my end. Thank you!
